### PR TITLE
1st stage of creation of User endpoints ==> cleanup and completion of CRUD of course with sufficient refactoring

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
**Description**

These changes have been made which will help to proceed with the integration and relationship handling of the user and course schemas and create the appropriate APIs for them

1). Full course code cleanup done 
2). Endpoints created along with refactoring of edit method
3).creation of delete by Id method
4).replacing @Query with @Param decorator in appropriate places and testing the response with postman
5).removal of irrelevant logic from controllers for cleaner code
6). Making appropriate changes and initialized endpoints with respect to its own set of unique data transfer object (dto)

**Type of Change:**

- [x] Code
- [x] New Feature

**Code/Quality Assurance Only**

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)

**How Has This Been Tested?**

npm start
Postman response check

**Checklist**

- [x] My code follows the code style of this project.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 please suggest changes (if any) on this one

File: src/models/course/course.controller.ts [hunk 1]
```diff
@@ -21,40 +17,33 @@ export class CourseController {
...
 
   // get selected course
   @Get('/:courseID')
-  async getSelectedCourse(@Res() res: Response, @Param('courseID') courseID) {
-    const course = await this.courseService.findCourseById(courseID);
-    if (!course) {
-      throw new NotFoundException('Selected course not found');
-    }
-    return res.status(HttpStatus.OK).json(course);
+  async getSelectedCourse(@Param('courseID') courseID: string) {
```
Comment (Abhishek-kumar09) [inline src/models/course/course.controller.ts:26]: `courseID` to `courseId`

File: src/models/course/course.controller.ts [hunk 2]
```diff
@@ -21,40 +17,33 @@ export class CourseController {
...
+    return await this.courseService.findCourseById(courseID);
   }
 
   // add a Course
   @Post('/create')
   async addCourse(@Body() courseDTO: CreateCourseDto) {
-    const course = await this.courseService.addCourse(courseDTO);
-    return course;
+    return await this.courseService.addCourse(courseDTO);
```
Comment (Abhishek-kumar09) [inline src/models/course/course.controller.ts:33]: `courseDto` to `createCourseDto`

File: src/models/course/course.controller.ts [hunk 3]
```diff
@@ -21,40 +17,33 @@ export class CourseController {
...
   }
 
   // update a course
-  @Put('/edit')
+  @Put('/edit/:courseID')
   async updateCourse(
-    @Query('id') cId: string,
+    @Param('courseID') courseID: string,
```
Comment (Abhishek-kumar09) [inline src/models/course/course.controller.ts:39]: `courseId`

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/models/course/course.controller.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/models/course/course.controller.ts]: fixed

File: src/models/course/course.controller.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/models/course/course.controller.ts]: fixed

File: src/models/course/course.controller.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/models/course/course.controller.ts]: fixed

**Pull request merged**